crm_rma_claim_make_claim/model/crm_claim_rma.py

```diff
@@ -0,0 +1,145 @@
+                    _('Error'),
+                    _('The claim client have claim supplier created.'))
+
+            partner_claim_line_ids = {}
+            # Grouping claim lines by partner (it is supplier of product)
+            for cl in claim_brw.claim_line_ids:
+                # If the claim line is in warranty, to do...
+                if cl.supplier_id.id in partner_claim_line_ids:
+                    partner_claim_line_ids.get(cl.supplier_id.id).append(cl.id)
+                else:
+                    partner_claim_line_ids[cl.supplier_id.id] = [cl.id]
+
+            for partner_id in partner_claim_line_ids:
+                for claim_line_id in partner_claim_line_ids.get(partner_id):
+
+                    # Search if have claim line have a child
+                    claim_lines_exists = claim_line_obj.search(
+                        [('claim_line_id', '=', claim_line_id)],)
+                    claim_lines_exists = [cla.id for cla in claim_lines_exists]
+
+                    # If claim_line_id does not have child
+                    if not claim_lines_exists:
+                        # The claim line of supplier claim is created
+                        claim_line_id = claim_line_obj.browse(claim_line_id)
+
+                        # invl = \
+                        #     invline_obj.search([('invoice_id',
+                        #                         '=',
+                        #                          claim_line_id.
+                        #                          supplier_invoice_id.id),
+                        #                        ('product_id',
+                        #                         '=',
+                        #                         claim_line_id.product_id.id
+                        #                         )])
+                        claim_line_new = \
+                            claim_line_id.copy({
+                                'claim_id': False,
+                                'number': '/',
+                                'claim_type':
+                                self.env.ref('crm_claim_rma_config.'
+                                             'crm_claim_type'
+                                             '_supplier').id,
+                                'claim_line_id': claim_line_id.id,
+                                # 'state': 'draft',
+                                # 'invoice_id': claim_line_id.
+                                # supplier_invoice_id.id,
+                                # 'supplier_invoice_id': inv_obj,
+                                # 'invoice_line_id': invl[0].id,
+                            })
+
+                        # is added for display in the view
+                        good_lines.append(claim_line_new.id)
+                    else:
+                        # Anyway. If claim line have child,
+                        # is added for display in the view
+                        good_lines += claim_lines_exists
```

Comment:
is it me or this for loop is really strange? It groups claim_lines by partner_id and then it completely ignores the grouping and just works on all claim lines normally..